src/frame/mod.rs

```diff
@@ -1,115 +1,82 @@
+    }
 }
-
-/// Frame Descriptor
-/// FLG     BD      (Content Size)  (Dictionary ID)     HC
-/// 1 byte  1 byte  0 - 8 bytes     0 - 4 bytes         1 byte
-#[allow(dead_code)]
-#[repr(C)]
-pub(crate) struct LZ4FFrameInfo {
-    pub content_size: Option<u64>,
-    pub block_size_id: BlockSize,
-    pub block_mode: BlockMode,
-    pub content_checksum_flag: Checksum,
-    // pub reserved: [u32; 5],
+impl From<io::Error> for Error {
+    fn from(e: io::Error) -> Self {
+        Error::IoError(e)
+    }
 }
 
-#[allow(dead_code)]
-impl LZ4FFrameInfo {
-    fn read(input: &[u8]) -> LZ4FFrameInfo {
-        // read flag bytes
-
-        let flg_byte = input[0];
-        assert!((flg_byte & 0b11000000) as usize == 1); // version is always 01
-
-        let block_mode = if (flg_byte & 1 << 5) as usize == 1 {
-            BlockMode::Independent
-        } else {
-            BlockMode::Linked
-        };
-        let content_checksum_flag = if (flg_byte & 1 << 2) as usize == 1 {
-            Checksum::ChecksumEnabled
-        } else {
-            Checksum::NoChecksum
-        };
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
 
-        // let content_size_included = (flg_byte & 1 << 4) == 1;
+impl std::error::Error for Error {}
 
-        LZ4FFrameInfo {
-            block_mode,
-            content_checksum_flag,
-            block_size_id: BlockSize::Default, // TODO
-            content_size: None,                // TODO
-        }
-    }
+/// Compress all bytes of `input`.
+pub fn compress(input: &[u8]) -> Vec<u8> {
+    compress_with(FrameInfo::default(), input)
+}
 
-    /// writes flag byte
-    ///
-    /// FLG byte
-    /// BitNumber   7-6         5                      4               3             2                 1           0
-    /// FieldName   Version     Block Independence     Block-Checksum  Content-Size  Content-Checksum  Reserved    DictID
-    ///
-    /// Content-Size (Optional) - The size of the uncompressed data included within the frame will be present as an 8 bytes unsigned little endian value, after the flags
-    fn write_flg_byte(&self) -> u32 {
-        let version = 1 << 6; // always 01
-        let mut res = version;
-        res |= (self.block_mode as u32) << 5;
-        if self.content_size.is_some() {
-            res |= 1 << 3; // set the content_size flag
-        }
-        res |= (self.content_checksum_flag as u32) << 2;
-        // res |= 0 << 4;
-        res
-    }
+/// Compress all bytes of `input`.
+pub fn compress_with(frame_info: FrameInfo, input: &[u8]) -> Vec<u8> {
+    let buffer = Vec::with_capacity(
+        header::MAX_FRAME_INFO_SIZE
+            + header::BLOCK_INFO_SIZE
+            + crate::block::compress::get_maximum_output_size(input.len()),
+    );
+    let mut enc = FrameEncoder::with_frame_info(frame_info, buffer);
+    enc.write_all(input).unwrap();
+    enc.finish().unwrap()
```

Comment:
I think the errors should be forwarded here

Reply:
I wasn't sure what to do here. In theory this will always succeed, as reading from bytes and writing to a vec always work.